### PR TITLE
cinnamon bulky

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Add bulky app for cinnamon desktop

~~Note: Currently fails to start with~~ On the cinnamon test vm everything works as it should

```
(.bulky-wrapped:3361228): GLib-GIO-CRITICAL **: 08:24:39.243: g_application_run: assertion 'G_IS_APPLICATION (application)' failed
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/cinnamon/bulky/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    gtk3
+    glib
+  ];
+
+  postPatch = ''
+    sed -e "s|/usr/share/locale|$out/share/locale|g"  \
+        -e "s|/usr/share/bulky|$out/share/bulky|g" \
+        -e "s|/usr/share/common-licenses|${common-licenses}/share/common-licenses|g" \
+        -e "s|__DEB_VERSION__|${version}|g" \
+        -i usr/lib/bulky/bulky.py
```
Comment (legendofmiracles) [inline pkgs/desktops/cinnamon/bulky/default.nix:43]: Couldn't this be replaced by a `substituteInPlace`?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125619` at 62e0b078 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>cinnamon.bulky</li>
    <li>common-licenses</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/data/misc/common-licenses/default.nix:15:3:
```
   |
15 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/data/misc/common-licenses/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchurl {
+    url = "http://deb.debian.org/debian/pool/main/b/base-files/base-files_${version}.tar.xz";
+    sha256 = "1i3hgd9vs14k819k441iibcgmi2zavnpqbnppyn2cz70kd830nbm";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share
+    cp -r licenses $out/share/common-licenses
+    cat debian/base-files.links | grep common-licenses | sed "s|usr|$out|g" | sed "s|^|ln -s |g" | sh -x
```
Comment (SuperSandro2000) [inline pkgs/data/misc/common-licenses/default.nix:18]:
```suggestion
    cat debian/base-files.links | grep common-licenses | sed -e "s|usr|$out|g" -e "s|^|ln -s |g" | bash -x
```

File: pkgs/desktops/cinnamon/bulky/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+stdenv.mkDerivation rec {
+  pname = "bulky";
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = "bulky";
+    rev = version;
+    sha256 = "NBlP10IM/+u8IRds4bdFyGWg3pJLRmlSLsdlndMVQqg=";
+    fetchSubmodules = false;
```
Comment (SuperSandro2000) [inline pkgs/desktops/cinnamon/bulky/default.nix:22]:
```suggestion
```
thats the default

File: pkgs/data/misc/common-licenses/default.nix (hunk 2 above)
Comment (legendofmiracles) [reply pkgs/data/misc/common-licenses/default.nix:18]: Shouldn't this maybe be runtimeShell instead? That way we can avoid pulling in more dependencies.

**Pull request merged**